service/raft/raft_address_map.hh

```diff
@@ -203,85 +137,57 @@ class raft_address_map {
 
     void drop_expired_entries() {
         auto list_it = _expiring_list.rbegin();
-        if (list_it == _expiring_list.rend()) {
-            return;
-        }
-        while (list_it->expired(_expiry_period)) {
-            auto base_list_it = list_it.base();
-            // When converting from rbegin() reverse_iterator to base iterator,
-            // we need to decrement it explicitly, because otherwise it will point
-            // to end().
-            --base_list_it;
+        while (list_it != _expiring_list.rend() && list_it->expired(_expiry_period)) {
             // Remove from both LRU list and base storage
-            unlink_and_dispose(to_set_iterator(base_list_it));
+            auto map_it = _map.find(list_it->entry_id());
```

Comment:
1. In which case we won't find it in the map?

2. OOP does not work very well here In my opinion, the deletion from `_expiring_list` is implicit, looking locally at this code it is not clear why it will happen. I would prefer to make `expiring_entry_ptr` a simple `struct` and remove it from the list explicitly. As a bonus, it would not be necessary to store `_expiring_list` in each `expiring_entry_ptr` - saving one pointer per entry.

Reply:
> In which case we won't find it in the map?

In none - this could indeed be an assertion.

Reply:
> I would prefer to make expiring_entry_ptr a simple struct and remove it from the list explicitly.

My preference is the opposite, I like how "being in the list" is an invariant of the object, it's a good usage of RAII -- constructor adds it to the list, destructor removes it, and all other methods maintain it. We don't have to manually manage the list every time we create or destroy these entries.

I do see the value in making the object smaller, but meh, there shouldn't be many objects of this type anyway, most of the entries should be non-expiring most of the time (and the total number of entries is still small).

Reply:
ok